src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -355,6 +355,7 @@ public async Task<TResult> WaitForCompletionAsync<TResult>(CancellationToken tok
                 catch (Exception e)
                 {
                     await StopProcessingUnderLock(e).ConfigureAwait(false);
+                    _additionalTask = Task.CompletedTask; // don't await the additional task if we had an error
                 }
 
                 await _additionalTask.ConfigureAwait(false);
```

Comment:
It seems we do not consider the token here maybe we also need something like this:
```
#if NET6_0_OR_GREATER
                    await result.WaitAsync(token).ConfigureAwait(false);
#else
                    await result.WithCancellation(token).ConfigureAwait(false);
#endif
```